### ISSUE TITLE
TagKey creation results in "googleapi: Error 500"

### ISSUE THREAD

**Issue by aschleck** [pulumi/pulumi-google-native#689]
### What happened?

I tried creating a TagKey and some associated TagValues. The TagKey could not be created due to failures in googleapi. I assume the TagValues would also fail for the same reason but was unable to get that far in this journey.

### Steps to reproduce

```
from pulumi_google_native.cloudresourcemanager import v3 as cloudresourcemanager
some_tag = cloudresourcemanager.TagKey("sometag", short_name="sometag", parent="organizations/23352352")
```

### Expected Behavior

I expected the tag to be created. Looking at the GraphQL query when creating a tag manually, it appears that the resource ID is not supplied by the client (because ultimately it will get a numeric ID like `tagKeys/2312314`.) My guess is that what's happening is Pulumi is providing an ID like `tagKeys/sometag-e167f02` based on the resource name to the API, and then the whole thing explodes.

```
{
  "requestContext": {
    ...
  },
  "query": "mutation CreateTagKey($key: IN_google_cloud_resourcemanager_v3_TagKey!) @Signature(bytes: \"1/gWcK8sOQnQ\") {\n  tagKeyService {\n    createTagKey(tagKey: $key) {\n      done\n      error {\n        code\n        message\n      }\n      metadata\n      name\n      response\n      result\n    }\n  }\n}",
  "variables": {
    "key": {
      "parent": "organizations/23352352",
      "shortName": "sometag",
      "description": null,
    }
  }
}
```

### Actual Behavior

```
Previewing update (prod):                                                                                                                                                                        [28/891]
     Type                                               Name               Plan       Info
     pulumi:pulumi:Stack                                machines-prod                 4 warnings
 +   └─ google-native:cloudresourcemanager/v3:TagKey    sometag          create

Diagnostics:
  pulumi:pulumi:Stack (machines-prod):                                                                                                                                                                       warning: cluster_id is deprecated: Deprecated. The name of the cluster. This field has been deprecated and replaced by the parent field.
    warning: cluster_id is deprecated: Deprecated. The name of the cluster. This field has been deprecated and replaced by the parent field.
    warning: cluster_id is deprecated: Deprecated. The name of the cluster. This field has been deprecated and replaced by the parent field.
    warning: cluster_id is deprecated: Deprecated. The name of the cluster. This field has been deprecated and replaced by the parent field.


Do you want to perform this update? details
  pulumi:pulumi:Stack: (same)
    [urn=urn:pulumi:prod::machines::pulumi:pulumi:Stack::machines-prod]
    + google-native:cloudresourcemanager/v3:TagKey: (create)
        [urn=urn:pulumi:prod::machines::google-native:cloudresourcemanager/v3:TagKey::sometag]
        [provider=urn:pulumi:prod::machines::pulumi:providers:google-native::default_0_25_0::8447cba3-acd2-46ff-8cd2-dc5f3c9ba986]
        name       : "tagKeys/sometag-0c379d0"
        parent     : "organizations/snip"
        shortName  : "sometag"

Do you want to perform this update? yes
Updating (prod):
     Type                                             Name           Status                  Info
     pulumi:pulumi:Stack                              machines-prod  **failed**              1 error; 4 warnings
 +   └─ google-native:cloudresourcemanager/v3:TagKey  sometag      **creating failed**     1 error

Diagnostics:
  pulumi:pulumi:Stack (machines-prod):
    warning: cluster_id is deprecated: Deprecated. The name of the cluster. This field has been deprecated and replaced by the parent field.
    warning: cluster_id is deprecated: Deprecated. The name of the cluster. This field has been deprecated and replaced by the parent field.
    warning: cluster_id is deprecated: Deprecated. The name of the cluster. This field has been deprecated and replaced by the parent field.
    warning: cluster_id is deprecated: Deprecated. The name of the cluster. This field has been deprecated and replaced by the parent field.
    error: update failed

  google-native:cloudresourcemanager/v3:TagKey (sometag):
    error: error sending request: googleapi: Error 500: Internal error encountered.: "https://cloudresourcemanager.googleapis.com/v3/tagKeys" map[__autonamed:true name:tagKeys/sometag-e167f02 parent:
organizations/snip shortName:sometag]

Resources:
    17 unchanged
```

(By the way the cluster_id warnings are because afaict `pulumi_google_native.container.v1.NodePool` both [requires](https://github.com/pulumi/pulumi-google-native/blob/880f14b8db136372fbdbc5058729dc8129dc9199/sdk/python/pulumi_google_native/container/v1/node_pool.py#L379) and [deprecates](https://github.com/pulumi/pulumi-google-native/blob/880f14b8db136372fbdbc5058729dc8129dc9199/sdk/python/pulumi_google_native/container/v1/node_pool.py#L382) `cluster_id`. I am not sure if this is intentional.)

### Output of `pulumi about`

```
CLI
Version      3.39.1
Go Version   go1.19
Go Compiler  gc

Plugins
NAME           VERSION
aws-native     0.24.0
google-native  0.25.0
python         unknown

Host
OS       ubuntu
Version  22.04
Arch     x86_64

This project is written in python: executable='<snip>' version='3.10.2
'

Current Stack: prod

TYPE                                        URN
pulumi:pulumi:Stack                         urn:pulumi:prod::machines::pulumi:pulumi:Stack::machines-prod
<snip>


Found no pending operations associated with prod

Backend
Name           bark
URL            <snip>
User           april
Organizations

Dependencies:
NAME                  VERSION
Mako                  1.2.1
pip                   22.0.3
pulumi-aws-native     0.24.0
pulumi-google-native  0.25.0
setuptools            60.9.3

Pulumi locates its logs in /tmp by default
warning: A new version of Pulumi is available. To upgrade from version '3.39.1' to '3.39.3', visit https://pulumi.com/docs/reference/install/ for manual instructions and release notes.
```

### Additional context

Thank you for working on this project!

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by stack72** [comment]
@viveklak / @lblackstone would love your thoughts here

**Comment by lblackstone** [comment]
This sounds like the provider's metadata for this resource is incorrect. The `cluster_id` being both required and deprecated also seems suspicious. Probably a problem with our code generation heuristics for this particular discovery document.

**Comment by viveklak** [comment]
The clusterId is marked required because of the `flatPath` for nodepools being set to `v1/projects/{projectsId}/locations/{locationsId}/clusters/{clustersId}/nodePools` in the discovery doc which makes clusterId a required field.

**Comment by viveklak** [comment]
Thanks for filing the issue. Indeed the reason for the failure was that the resource was being autonamed but it seems the service automatically generates the name instead. The fix is in #695. I was able to create tag keys and values with the change.
